launcher/internal/gpu/driverinstaller.go

```
@@ -208,20 +236,55 @@ func setGPUStateToReady() error {
 	return nil
 }
 
-func isGPUCCModeEnabled(logger logging.Logger, gpuType deviceinfo.GPUType) (bool, error) {
-	// Run nvidia-smi conf-compute command to check if confidential compute mode is ON.
+func isGPUCCModeEnabled() (bool, error) {
+	ccMode, err := QueryCCMode()
+	if err != nil {
+		return false, err
+	}
+	return ccMode == CCModeON, nil
+}
+
+// QueryCCMode executes nvidia-smi to determine the current Confidential Computing (CC) mode status of the GPU.
+// If DEVTOOLS mode is enabled, it would override CC mode as DEVTOOLS. DEVTOOLS mode would be enabled only when CC mode is ON.
+func QueryCCMode() (CCMode, error) {
+	ccMode := CCModeOFF
+	// Run nvidia-smi conf-compute command to get the confidential computing mode status.
 	nvidiaSmiCmd := fmt.Sprintf("%s/bin/nvidia-smi", InstallationHostDir)
 	ccModeOutput, err := exec.Command(nvidiaSmiCmd, "conf-compute", "-f").Output()
+	if err != nil {
+		return "", err
+	}
+
 	// The nvidia-smi conf-compute command fails for GPU which doesn't support confidential computing.
 	// This check would bypass nvidia-smi conf-compute command for GPU not having confidential compute support.
 	if strings.Contains(string(ccModeOutput), "No CC capable devices found") {
-		logger.Info(fmt.Sprintf("Confidential Computing is not supported for GPU type : %s", gpuType.String()))
-		return false, nil
+		logger.Info("Confidential Computing is not supported for GPU")
+		return ccMode, nil
```

Should there be an UNSUPPORTED value?

---

I have updated it to fail early for non confidential GPUs and so this would be no longer required. Additionally, I updated it to use [this enum](https://github.com/google/go-tpm-tools/blob/cs_cgpu_h100/proto/attest.proto#L195) which would provide UNSET status for such case.